src/persist-client/src/read.rs

```diff
@@ -138,13 +141,16 @@ where
             // for retries.
             //
             // TODO: Restructure this loop so this is more obviously correct.
-            assert_eq!(self.batches.pop().as_ref(), Some(&key));
+            assert_eq!(
+                self.batches.pop().expect("known to exist"),
+                (key.clone(), desc.clone())
```

Comment:
I don't like the clones here. The other option would be to also destructure the result of the `pop()`, to sth like `(key1, desc1)` and assert individually, wdyt?

Reply:
destructure the pop sgtm. I wouldn't care about the clone if it was a debug_assert, but given that it's not I agree that it's worth addressing the clone